### PR TITLE
FEX118 Add logo and potentially link to great profile

### PR DESCRIPTION
**Pull request by benpbenp**
- If a company exists in GREAT find-a-supplier, then show the GREAT logo
- If additionally the company has a published profile, and they have a companies house id, link to it.

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ let config = {
   paginationMaxResults: 10000,
   paginationOffset: 20,
   daysUntilDissolvedCompaniesDeleted: 180,
+  findASupplierProfileUrlPrefix: 'https://www.great.gov.uk/trade/suppliers/',
```
Comment (yozzo) [inline config/index.js]: needs to come from `.env`

File: src/app/transformers.js [hunk 2]
```diff
@@ -146,19 +146,35 @@ function transformPostcodeFilter (string) {
...
  *         and saved to company.days_to_deletion.
  *     OTOH if company.time_since_remove_recorded is not valid,
  *     it is set to null.
+ * @param company
+ * @returns {*}
+ * @private
  */
 
-function prepareCompany (company) {
+function _prepareCompanyDissolvedInfo (company) {
```
Comment (yozzo) [inline src/app/transformers.js]: would the underscore denote a private function? We don't use this notation in .js

File: src/app/components/render-collection/template.njk [hunk 3]
```diff
@@ -158,8 +160,20 @@
...
                             <div class="govuk-grid-column-full">
 
-                                {% if item.line_of_business or item.dit_sectors  %}
+                                {% if item.line_of_business or item.dit_sectors or item.is_joined_find_a_supplier %}
                                     <div class="govuk-!-display-block govuk-!-padding-top-1 govuk-!-padding-bottom-4">
+                                        {% if item.is_joined_find_a_supplier %}
+                                            <div class="govuk-!-display-block great-link-with-logo">
+                                                <img src="/public/img/great.svg" alt="GREAT" class="great-logo"/>
+                                                {% if item.find_a_supplier_url %}
+                                                <a href="{{ item.find_a_supplier_url }}">
```
Comment (yozzo) [inline src/app/components/render-collection/template.njk]: indent one tab everything inside the loop

File: config/index.js (hunk 1 above)
Comment (benpbenp) [reply config/index.js]: ok

File: src/app/transformers.js (hunk 2 above)
Comment (benpbenp) [reply src/app/transformers.js]: ok

File: src/app/components/render-collection/template.njk (hunk 3 above)
Comment (benpbenp) [reply src/app/components/render-collection/template.njk]: ok

File: src/app/transformers.js [hunk 4]
```diff
@@ -172,6 +188,49 @@ function prepareCompany (company) {
...
+ * comma separated string
+ * 2. Sets the find_a_supplier_url using a templated GREAT url plus the company's
+ *  national identification number, if it has one and if company.is_published_find_a_supplier
+ *
+ * @param company
+ * @returns {*}
+ * @private
+ */
+
+function prepareCompanyGreatProfile (company) {
```
Comment (yeahfro) [inline src/app/transformers.js]: If `company` was destructured it could make the function content look cleaner

File: src/app/transformers.js [hunk 5]
```diff
@@ -146,19 +146,35 @@ function transformPostcodeFilter (string) {
...
- * Transformations applied:
- *  1. If company.time_since_remove_recorded is present and valid,
+ * zeroFill, for a givin input `number` (could be a number or a string),
+ * and a given `width`, pad the number with enough leading zeros so that
+ * it is of width `width` and return the result.
+ * @param number
+ * @param width
+ * @returns {string}
+ */
+function zeroFill (number, width) {
```
Comment (yeahfro) [inline src/app/transformers.js]: Does Lodash's `padStart` achieve the same thing?

Comment (benpbenp) [reply src/app/transformers.js]: why yes, yes it is.

File: src/app/transformers.js (hunk 4 above)
Comment (benpbenp) [reply src/app/transformers.js]: @yozzo Is this what you were saying on slack too? Do yall mean something like 
```
const { is_published_find_a_supplier, service_usage, national_identification_number,
           national_identification_system_code } = company
```

at the beginning of the function?

**Pull request merged**